### ISSUE TITLE
Fix crash when lifecycleEvents is enabled on Tracker initialisation

### ISSUE THREAD

**Issue by shumidub** [snowplow/snowplow-android-tracker#339]
On create Tracker class instance by private constructor Tracker(Tracker.TrackerBuilder builder) called this block of code 

```
if ((this.lifecycleEvents || this.sessionContext) && VERSION.SDK_INT >= 14) {
            Executor.execute(new Runnable() {
                public void run() {
                    ProcessLifecycleOwner.get().getLifecycle().addObserver(new ProcessObserver());
                }
            });
}
```

method addObserver(new ProcessObserver()) called on background thread.

But Lifecycle.addObserver(observer) method has annotation  _**"@MainThread"**_ and should be invoced only from Main thread.

By this reason, now our application have many crashes as bellow:

```
"Caused by java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
       at java.util.ArrayList.remove(ArrayList.java:503)
       at androidx.lifecycle.LifecycleRegistry.popParentState(LifecycleRegistry.java:195)
       at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:294)
       at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
       at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:189)
"
```

Please fix it and call addObserver() method only on MainThread

**Comment by shumidub** [comment]
What is the time estimate for fix ?

**Comment by paulboocock** [comment]
Hi @shumidub 
Thank you for the issue report. We are looking into getting a fix ready for this in the next release of the Android Tracker.
We will keep you up to date with our progress.

**Comment by paulboocock** [comment]
@shumidub this is now released in 1.3.3 here: https://bintray.com/snowplow/snowplow-maven/snowplow-android-tracker
